main.tf

```
@@ -2,7 +2,7 @@ resource "aws_elasticache_replication_group" "redis" {
   engine = var.global_replication_group_id == null ? "redis" : null
 
   parameter_group_name = var.global_replication_group_id == null ? aws_elasticache_parameter_group.redis.name : null
-  subnet_group_name    = aws_elasticache_subnet_group.redis.name
+  subnet_group_name    = aws_elasticache_subnet_group.redis[0].name
```

```suggestion
  subnet_group_name    = try(var.subnet_group_name, aws_elasticache_subnet_group.redis[0].name)
```
https://developer.hashicorp.com/terraform/language/functions/try

---

[9b9558b](https://github.com/StuDocu/terraform-aws-elasticache-redis/pull/5/commits/9b9558b5f24bd89a088a2ecdf632d7e1eb9d8de9)